pkg/antctl/raw/featuregates/command_test.go

```diff
@@ -0,0 +1,345 @@
+			response:       agentResponse,
+		},
+		{
+			name:           "get featuregates in controller Pod",
+			runE:           controllerLocalRunE,
+			expectedOutput: "Antrea Agent Feature Gates\nFEATUREGATE              STATUS         VERSION   \nEndpointSlice            Disabled       ALPHA     \nAntreaIPAM               Disabled       ALPHA     \nFlowExporter             Disabled       ALPHA     \nNetworkPolicyStats       Enabled        BETA      \nAntreaPolicy             Enabled        BETA      \nMulticast                Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nAntreaProxy              Enabled        BETA      \nEgress                   Enabled        BETA      \nNodePortLocal            Enabled        BETA      \nMulticluster             Enabled        ALPHA     \nAntrea Controller Feature Gates\nFEATUREGATE              STATUS         VERSION   \nAntreaPolicy             Enabled        BETA      \nNodeIPAM                 Disabled       ALPHA     \nTraceflow                Enabled        BETA      \nServiceExternalIP        Disabled       ALPHA     \nEgress                   Enabled        BETA      \nNetworkPolicyStats       Enabled        BETA      \n",
+			response:       fullResponse,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeHttpClient := fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+				return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(bytes.NewBuffer(tt.response))}, nil
+			})
+			restClient.Client = fakeHttpClient
+			testClients.restClient = restClient
+
+			buf := new(bytes.Buffer)
+			Command.SetOutput(buf)
+			Command.SetOut(buf)
+			Command.SetErr(buf)
+
+			err := tt.runE(Command, nil)
+			assert.NoError(t, err)
+			assert.Equal(t, tt.expectedOutput, buf.String())
+		})
+	}
+}
+
+func TestGetConfigAndClients(t *testing.T) {
+	fakeConfigs := []byte(`apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: data
+    server: https://localhost
+  name: fake-cluster
+contexts:
+- context:
+    cluster:  fake-cluster
+    user:  user-id
+  name:  fake-cluster
+current-context:  fake-cluster
+kind: Config`)
+
+	var err error
+	fakeKubeconfig, err := os.CreateTemp("", "fakeKubeconfig")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(fakeKubeconfig.Name())
+	fakeKubeconfig.Write(fakeConfigs)
+	kubeconfig := ""
+	Command.Flags().StringVarP(&kubeconfig, "kubeconfig", "k", fakeKubeconfig.Name(), "path of kubeconfig")
+	_, _, _, err = getConfigAndClients(Command)
```

Comment:
It's not clear to me what we are testing here. I assume the clients are not "valid" given that we use a fake kubeconfig?

Reply:
The purpose to check the function can resolve config file correctly. I refined a little bit and add some new checks, could you check? thanks.